src/CodeGen_LLVM.cpp

```diff
@@ -2021,6 +2021,7 @@ void CodeGen_LLVM::scalarize(Expr e) {
 }
 
 void CodeGen_LLVM::codegen_predicated_vector_store(const Store *op) {
+    internal_assert(!op->is_atomic); // Doesn't support atomic vector store
```

Comment:
Added a user_assert.
Scalarizing the store sounds like a reasonable plan. I can code it up later.

Reply:
Added vectorization through scalarization of the store. Predicated vector is still not supported since it requires pretty complicated codegen logic for the CAS loop (need to specialize when the predicate does recursively depend on the function).